File: src/BaseHmiTypes/Screens/Controls/HmiAlarmStatusBarPanel.cs
using BaseHmiTypes.Common;
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Controls;

public sealed class HmiAlarmStatusBarPanel
{
    public HmiAlarmStatusBarPanelType Type { get; set; }

    public string? SourceType { get; set; }

    public HmiProperty<bool>? Visible { get; set; }

    public HmiProperty<int>? Order { get; set; }

    public HmiMultilingualText? Tooltip { get; set; }
}

File: typescript/base-hmi-types/src/screens/controls/HmiAlarmStatusBarPanel.ts
import { HmiMultilingualText } from "../../common/HmiMultilingualText.js";
import { HmiProperty } from "../base.js";
import { HmiAlarmStatusBarPanelType } from "./HmiAlarmStatusBarPanelType.js";

export class HmiAlarmStatusBarPanel {
  type = HmiAlarmStatusBarPanelType.Unknown;
  sourceType?: string;
  visible?: HmiProperty<boolean>;
  order?: HmiProperty<number>;
  tooltip?: HmiMultilingualText;
}
